File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmA3Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import systems.zlink.e2e.spotservice.client.Support.GatewayScenarioClient;

public final class SmA3Scenario {
    private SmA3Scenario() {
    }

    public static void run(GatewayScenarioClient gateway) {
        gateway.runMode("owner");
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Client/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/client/scenarios/SmA3Scenario.kt
package systems.zlink.e2e.kotlin.spotservice.client.scenarios

import systems.zlink.contracts.core.RoutingId
import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.client.support.REQUEST_TIMEOUT
import systems.zlink.e2e.kotlin.spotservice.client.support.ensure
import systems.zlink.e2e.kotlin.spotservice.client.support.eventually
import systems.zlink.framework.spots.ZLinkSpotOutbound

internal object SmA3Scenario {
    fun run(outbound: ZLinkSpotOutbound) {
        val roomA = eventually {
            outbound.requestToSpot(
                RoutingId.from("room-a"),
                Contracts.StateReq("owner-a"),
            )
                .timeout(REQUEST_TIMEOUT)
                .await(Contracts.StateRes::class.java)
        }
        ensure(roomA.nodeRid == "play-a", "SM-A3 room-a owner mismatch")
        println("scenario SM-A3 passed")
    }
}
